### DISCUSSION TITLE
Tree Component?

### DISCUSSION THREAD

**Discussion by corysimmons** [shadcn-ui/ui#2329] [Ideas]
I'd love it if shadcn-ui had a Tree component. I frequently find myself looking for this kind of component.

Prior art:
- https://mui.com/x/react-tree-view/
- https://github.com/brimdata/react-arborist
- https://ant.design/components/tree
- https://dgreene1.github.io/react-accessible-treeview/ (this one looks new/accessible/maintained/clean, although it might lack some features?)
- https://primereact.org/tree/
- https://github.com/frontend-collective/react-sortable-tree
- a billion more

~~Actually this one looks really good at first glance https://jakezatecky.github.io/react-checkbox-tree/~~

This one seems to have everything https://primereact.org/tree/

**Comment by nagaozen** [comment]
FYI, this discussion seems to be related to feature highlighted here: https://github.com/shadcn-ui/ui/issues/355#issuecomment-1703767574

**Comment by dingshaohua-com** [comment]
i need too

**Comment by nthomsencph** [comment]
please

**Comment by thishumanoid** [comment]
yeah please bring this
